src/integer.c

```diff
@@ -2225,6 +2225,42 @@ Obj FuncGCD_INT ( Obj self, Obj opL, Obj opR )
 }
 
 
+/****************************************************************************
+**
+*/
+Obj FactorialInt(Obj opN)
+{
+    if (!IS_INTOBJ(opN))
+        ErrorMayQuit("<n> must be a small integer", 0L, 0L);
+
+    Int n = INT_INTOBJ(opN);
+    if (n < 0)
+        ErrorMayQuit("<n> must be nonnegative", 0L, 0L);
+
+    mpz_t mpzResult;
+    mpz_init(mpzResult);
+    mpz_fac_ui(mpzResult, n);
+
+    // convert mpzResult into a GAP integer object.
+    Obj result = GMPorINTOBJ_MPZ(mpzResult);
+
+    // free mpzResult
+    mpz_clear(mpzResult);
+
+    return result;
+}
+
+
+/****************************************************************************
+**
+*/
+Obj FuncFACTORIAL_INT(Obj self, Obj n)
+{
+    REQUIRE_INT_ARG("FactorialInt", "n", n);
```

Comment:
Shouldn't this be `Factorial` instead of `FactorialInt` (which does not exist on GAP level)?

Reply:
Yes, that should be `Factorial` and not `FactorialInt`.

The check is in a separate function because so far all those integer functions had a "kernel version" (i.e. `FactorialInt`), and a "library version (i.e. `FuncFACTORIAL_INT`) -- in other words, blind habit. But its unlikely that we will want to directly call `Factorial` from the kernel, so that's overkill here, so I agree we can merge the two together.

(These remarks apply similarly to `Binomial` and a few others, I'll look into tweaking those, too)

Reply:
The reason why I did not remove `InstallGlobalFunction(Factorial, FACTORIAL_INT);` is that then I also need to remove the `Declare*` counterpart, and decide where to move the GAPDoc comment; and I'd rather keep the change as small as possible.